<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot a GPos or GRanges Object Using Shiny and Gosling — gosling_plot • AlphaMissenseR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot a GPos or GRanges Object Using Shiny and Gosling — gosling_plot"><meta name="description" content="plot_granges() creates a Shiny app that displays a
Gosling plot of a given GRanges object.  It visualizes genomic
ranges with both rectangle and point representations, and
allows for customization of the plot title and subtitle."><meta property="og:description" content="plot_granges() creates a Shiny app that displays a
Gosling plot of a given GRanges object.  It visualizes genomic
ranges with both rectangle and point representations, and
allows for customization of the plot title and subtitle."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">AlphaMissenseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/introduction.html">A. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/alphafold.html">B. AlphaFold Integration</a></li>
    <li><a class="dropdown-item" href="../articles/clinvar.html">C. ClinVar Integration</a></li>
    <li><a class="dropdown-item" href="../articles/benchmarking.html">D. Benchmarking with ProteinGym</a></li>
    <li><a class="dropdown-item" href="../articles/issues.html">E. Issues &amp; Solutions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mtmorgan/AlphaMissenseR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Plot a GPos or GRanges Object Using Shiny and Gosling</h1>
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/AlphaMissenseR/blob/HEAD/R/Gosling.R" class="external-link"><code>R/Gosling.R</code></a></small>
      <div class="d-none name"><code>Gosling.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><code>plot_granges()</code> creates a Shiny app that displays a
Gosling plot of a given GRanges object.  It visualizes genomic
ranges with both rectangle and point representations, and
allows for customization of the plot title and subtitle.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gosling_plot</span><span class="op">(</span></span>
<span>  <span class="va">granges</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"GRanges Plot"</span>,</span>
<span>  subtitle <span class="op">=</span> <span class="st">"Stacked nucleotide example"</span>,</span>
<span>  plot_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bar"</span>, <span class="st">"lollipop"</span><span class="op">)</span>,</span>
<span>  create_app <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-granges">granges<a class="anchor" aria-label="anchor" href="#arg-granges"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges::GRanges</a></code> (e.g.,
<code><a href="https://rdrr.io/pkg/GenomicRanges/man/GPos-class.html" class="external-link">GenomicRanges::GPos</a></code>) object containing the genomic ranges
to be plotted.</p></dd>


<dt id="arg-title">title<a class="anchor" aria-label="anchor" href="#arg-title"></a></dt>
<dd><p>character(1) Title of the plot. Default is
"GRanges Plot".</p></dd>


<dt id="arg-subtitle">subtitle<a class="anchor" aria-label="anchor" href="#arg-subtitle"></a></dt>
<dd><p>character(1) The subtitle of the plot. Default is
"Stacked nucleotide example".</p></dd>


<dt id="arg-plot-type">plot_type<a class="anchor" aria-label="anchor" href="#arg-plot-type"></a></dt>
<dd><p>character(1) Select the type of gosling plot.
Default is "bar"; available types are described in Details.</p></dd>


<dt id="arg-create-app">create_app<a class="anchor" aria-label="anchor" href="#arg-create-app"></a></dt>
<dd><p>logical(1) Produce a Shiny app for Gosling
visualization. Default <code>TRUE</code> when used interactively.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p><code>gosling_plot()</code> with <code>create_app = TRUE</code> returns a
<code>shinyApp</code> object that, when run, displays the Gosling
plot. When <code>create_app = FALSE</code>, the (invisible) return value
represents the gosling object to be used in Shiny apps.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function supports 2 types of plots as selected through the
<code>plot_type</code> argument: (1) <code>"bar"</code>: a barplot-like view of the
pathogenicity score at each position, similar to a sequencing
coverage plot, and (2) <code>"lollipop"</code>: a lollipop plot focusing on
the pathogenicity classification (ambiguous, benign, pathogenic) at
each position. It requires that the <code><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GenomicRanges::GRanges</a></code>
object has the following metadata columns: 'am_class' (effect
classification),'am_pathogenicity' (pathogenicity score), 'ALT'
(alternative allele) and 'REF' (reference allele).</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>This function requires the <code>shiny</code>, <code>shiny.gosling</code>, and
<code>GenomicRanges</code> packages to be installed.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">## Create a sample GRanges object from AlphamissenseR</span></span></span>
<span class="r-in"><span><span class="va">gpos</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="AlphaMissense.html">am_data</a></span><span class="op">(</span><span class="st">"hg38"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">uniprot_id</span> <span class="op">==</span> <span class="st">"Q1W6H9"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="GenomicRanges.html">to_GPos</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Plot `gpos` using shiny.gosling</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"shiny.gosling"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>   <span class="fu">gosling_plot</span><span class="op">(</span></span></span>
<span class="r-in"><span>       <span class="va">gpos</span>, title <span class="op">=</span> <span class="st">"Q1W6H_track"</span>, subtitle <span class="op">=</span> <span class="st">"bar plot example"</span>,</span></span>
<span class="r-in"><span>       plot_type <span class="op">=</span> <span class="st">"bar"</span>, create_app <span class="op">=</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span>   <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Martin Morgan, Tram Nguyen, Chan Zuckerberg Initiative DAF CZF2019-002443, NIH NCI ITCR U24CA180996, NIH NCI IOTN U24CA232979, NIH NCI ARTNet U24CA274159.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

