<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="AlphaMissenseR">
<title>Accessing AlphaMissense Resources in R • AlphaMissenseR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Accessing AlphaMissense Resources in R">
<meta property="og:description" content="AlphaMissenseR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">AlphaMissenseR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.21</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/a_introduction.html">Accessing AlphaMissense Resources in R</a>
    <a class="dropdown-item" href="../articles/b_visualization.html">Visualization</a>
    <a class="dropdown-item" href="../articles/c_issues.html">Issues &amp; Solutions</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/mtmorgan/AlphaMissenseR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Accessing AlphaMissense Resources in R</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/mtmorgan/AlphaMissenseR/blob/HEAD/vignettes/a_introduction.Rmd" class="external-link"><code>vignettes/a_introduction.Rmd</code></a></small>
      <div class="d-none name"><code>a_introduction.Rmd</code></div>
    </div>

    
    
<p>Original version: 26 September, 2023</p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The AlphaMissense <a href="https://www.science.org/doi/epdf/10.1126/science.adg7492" class="external-link">publication</a>
outlines how a variant of AlphaFold / DeepMind was used to predict
missense variant pathogenicity. Supporting data on <a href="https://zenodo.org//record/8360242" class="external-link">Zenodo</a> include, for
instance 70+M variants across hg19 and hg38 genome builds. The
AlphaMissense package allows ready access to the data, downloading
individual files to DuckDB databases for ready exploration and
integration into <em>R</em> and <em>Bioconductor</em> workflows.</p>
<p>Install the package from <em>Bioconductor</em> or GitHub, ensuring
correct <em>Bioconductor</em> dependencies.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="st">"BiocManager"</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/installed.packages.html" class="external-link">installed.packages</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, repos <span class="op">=</span> <span class="st">"https://cran.r-project.org"</span><span class="op">)</span></span></code></pre></div>
<p>When the package is available on <em>Bioconductor</em>, use</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/version.html" class="external-link">version</a></span><span class="op">(</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="st">"3.19"</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"AlphaMissenseR"</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span></span>
<span>        <span class="st">"'AlphaMissenseR' requires Bioconductor version 3.19 or later, "</span>,</span>
<span>        <span class="st">"install from GitHub?"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Use the pre-release or devel version with</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span></span>
<span>    <span class="st">"mtmorgan/AlphaMissenseR"</span>,</span>
<span>    repos <span class="op">=</span> <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/repositories.html" class="external-link">repositories</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Load the library.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://mtmorgan.github.io/AlphaMissenseR/">AlphaMissenseR</a></span><span class="op">)</span></span></code></pre></div>
<p>Learn about available data by visiting the Zenodo record where data
will be retrieved from.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/AlphaMissense.html">am_browse</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="discovery-retrieval-and-use">Discovery, retrieval and use<a class="anchor" aria-label="anchor" href="#discovery-retrieval-and-use"></a>
</h2>
<p>Use <code><a href="../reference/AlphaMissense.html">am_available()</a></code> to discover data resources available
for representation in DuckDB databases. The <code>cached</code> column
is initially <code>FALSE</code> for all data sets; <code>TRUE</code>
indicates that a data set has been downloaded by <code><a href="../reference/AlphaMissense.html">am_data()</a></code>,
as described below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/AlphaMissense.html">am_available</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 6</span></span></span>
<span><span class="co">#&gt;   record   key                             size cached filename            link </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 10813168 gene_hg38                     <span style="text-decoration: underline;">253</span>636 TRUE   AlphaMissense_gene… http…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 10813168 isoforms_hg38             <span style="text-decoration: underline;">1</span>177<span style="text-decoration: underline;">361</span>934 FALSE  AlphaMissense_isof… http…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 10813168 isoforms_aa_substitutions <span style="text-decoration: underline;">2</span>461<span style="text-decoration: underline;">351</span>945 FALSE  AlphaMissense_isof… http…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 10813168 hg38                       642<span style="text-decoration: underline;">961</span>469 TRUE   AlphaMissense_hg38… http…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> 10813168 hg19                       622<span style="text-decoration: underline;">293</span>310 FALSE  AlphaMissense_hg19… http…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> 10813168 gene_hg19                     <span style="text-decoration: underline;">243</span>943 FALSE  AlphaMissense_gene… http…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> 10813168 aa_substitutions          <span style="text-decoration: underline;">1</span>207<span style="text-decoration: underline;">278</span>510 TRUE   AlphaMissense_aa_s… http…</span></span></code></pre></div>
<p>The available data sets use the most recent record as of 25
September, 2023; this can be changed by specifying an alternative as the
<code>record=</code> argument or changed globally by setting an
environment variable <code>ALPHAMISSENSE_RECORD</code> <em>before</em>
loading the package.</p>
<p>Use <code><a href="../reference/AlphaMissense.html">am_data()</a></code> to download a data resource and store it
in a DuckDB database. The <code>key=</code> argument is from the column
returned by <code><a href="../reference/AlphaMissense.html">am_available()</a></code>. Files are cached locally (using
<a href="https://bioconductor.org/packages/BiocFileCache" class="external-link">BiocFileCache</a>)
so this operation is expensive only the first time. Each
<code>record=</code> is stored in a different database.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AlphaMissense.html">am_data</a></span><span class="op">(</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="va">tbl</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;hg38&gt; [?? x 10]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;    CHROM   POS REF   ALT   genome uniprot_id transcript_id     protein_variant</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1  <span style="text-decoration: underline;">69</span>094 G     T     hg38   Q8NH21     ENST00000335137.4 V2L            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1  <span style="text-decoration: underline;">69</span>094 G     C     hg38   Q8NH21     ENST00000335137.4 V2L            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1  <span style="text-decoration: underline;">69</span>094 G     A     hg38   Q8NH21     ENST00000335137.4 V2M            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1  <span style="text-decoration: underline;">69</span>095 T     C     hg38   Q8NH21     ENST00000335137.4 V2A            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1  <span style="text-decoration: underline;">69</span>095 T     A     hg38   Q8NH21     ENST00000335137.4 V2E            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1  <span style="text-decoration: underline;">69</span>095 T     G     hg38   Q8NH21     ENST00000335137.4 V2G            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1  <span style="text-decoration: underline;">69</span>097 A     G     hg38   Q8NH21     ENST00000335137.4 T3A            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1  <span style="text-decoration: underline;">69</span>097 A     C     hg38   Q8NH21     ENST00000335137.4 T3P            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1  <span style="text-decoration: underline;">69</span>097 A     T     hg38   Q8NH21     ENST00000335137.4 T3S            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1  <span style="text-decoration: underline;">69</span>098 C     A     hg38   Q8NH21     ENST00000335137.4 T3N            </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: am_pathogenicity &lt;dbl&gt;, am_class &lt;chr&gt;</span></span></span></code></pre></div>
<p>The return value <code>tbl</code> is a table from a DuckDB database.
A (read-only) connection to the database itself is available with</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/db.html">db_connect</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>This connection remains open throughout the session; call
<code>db_disconnect(db)</code> or <code><a href="../reference/db.html">db_disconnect_all()</a></code> to
close it at the end of the session.</p>
<p>The database contains tables for each key downloaded. As an
alternative to <code><a href="../reference/AlphaMissense.html">am_available()</a></code> / <code><a href="../reference/AlphaMissense.html">am_data()</a></code>,
view available tables and create a <a href="https://cran.r-project.org/package=dplyr" class="external-link">dplyr</a> / <a href="https://cran.r-project.org/package=dbplyr" class="external-link">dbplyr</a> tibble of
the table of interest.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/db.html">db_tables</a></span><span class="op">(</span><span class="va">db</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "aa_substitutions" "gene_hg38"        "hg38"</span></span>
<span></span>
<span><span class="va">tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/tbl.html" class="external-link">tbl</a></span><span class="op">(</span><span class="va">db</span>, <span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="va">tbl</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;hg38&gt; [?? x 10]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;    CHROM   POS REF   ALT   genome uniprot_id transcript_id     protein_variant</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr1  <span style="text-decoration: underline;">69</span>094 G     T     hg38   Q8NH21     ENST00000335137.4 V2L            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr1  <span style="text-decoration: underline;">69</span>094 G     C     hg38   Q8NH21     ENST00000335137.4 V2L            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr1  <span style="text-decoration: underline;">69</span>094 G     A     hg38   Q8NH21     ENST00000335137.4 V2M            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr1  <span style="text-decoration: underline;">69</span>095 T     C     hg38   Q8NH21     ENST00000335137.4 V2A            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr1  <span style="text-decoration: underline;">69</span>095 T     A     hg38   Q8NH21     ENST00000335137.4 V2E            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr1  <span style="text-decoration: underline;">69</span>095 T     G     hg38   Q8NH21     ENST00000335137.4 V2G            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr1  <span style="text-decoration: underline;">69</span>097 A     G     hg38   Q8NH21     ENST00000335137.4 T3A            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr1  <span style="text-decoration: underline;">69</span>097 A     C     hg38   Q8NH21     ENST00000335137.4 T3P            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr1  <span style="text-decoration: underline;">69</span>097 A     T     hg38   Q8NH21     ENST00000335137.4 T3S            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr1  <span style="text-decoration: underline;">69</span>098 C     A     hg38   Q8NH21     ENST00000335137.4 T3N            </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: am_pathogenicity &lt;dbl&gt;, am_class &lt;chr&gt;</span></span></span></code></pre></div>
<p>It is fast and straight-forward to summarize the data, e.g., the
number of variants assigned to each pathogenicity class.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">am_class</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [3 x 2]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;   am_class                 n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> likely_benign     40<span style="text-decoration: underline;">917</span>351</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ambiguous          8<span style="text-decoration: underline;">009</span>648</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> likely_pathogenic 22<span style="text-decoration: underline;">770</span>557</span></span></code></pre></div>
<p>Or the average pathogenicity score in each class…</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">am_class</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, pathogenecity <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">am_pathogenicity</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [3 x 3]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;   am_class                 n pathogenecity</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> likely_benign     40<span style="text-decoration: underline;">917</span>351         0.147</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> likely_pathogenic 22<span style="text-decoration: underline;">770</span>557         0.863</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ambiguous          8<span style="text-decoration: underline;">009</span>648         0.444</span></span></code></pre></div>
<p>Or the number of transitions between <code>REF</code> and
<code>ALT</code> nucleotides across all variants.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">REF</span>, <span class="va">ALT</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="st">"ALT"</span>, values_from <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"REF"</span>, <span class="st">"A"</span>, <span class="st">"C"</span>, <span class="st">"G"</span>, <span class="st">"T"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">REF</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:     SQL [4 x 5]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database:   DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Ordered by: REF</span></span></span>
<span><span class="co">#&gt;   REF         A       C       G       T</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> A          <span style="color: #BB0000;">NA</span> 6<span style="text-decoration: underline;">420</span>083 5<span style="text-decoration: underline;">449</span>617 5<span style="text-decoration: underline;">953</span>590</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> C     6<span style="text-decoration: underline;">183</span>248      <span style="color: #BB0000;">NA</span> 6<span style="text-decoration: underline;">963</span>298 4<span style="text-decoration: underline;">909</span>869</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> G     4<span style="text-decoration: underline;">894</span>493 6<span style="text-decoration: underline;">937</span>258      <span style="color: #BB0000;">NA</span> 6<span style="text-decoration: underline;">161</span>594</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> T     5<span style="text-decoration: underline;">955</span>039 5<span style="text-decoration: underline;">445</span>540 6<span style="text-decoration: underline;">423</span>927      <span style="color: #BB0000;">NA</span></span></span></code></pre></div>
<p>It is straight-forward to select variants in individual regions of
interest, e.g., the first 200000 nucleoties of chromosome 4.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CHROM</span> <span class="op">==</span> <span class="st">"chr4"</span>, <span class="va">POS</span> <span class="op">&gt;</span> <span class="fl">0</span>, <span class="va">POS</span> <span class="op">&lt;=</span> <span class="fl">200000</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [?? x 10]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;    CHROM   POS REF   ALT   genome uniprot_id transcript_id     protein_variant</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr4  <span style="text-decoration: underline;">59</span>430 G     C     hg38   Q8IYB9     ENST00000610261.6 E2Q            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr4  <span style="text-decoration: underline;">59</span>430 G     A     hg38   Q8IYB9     ENST00000610261.6 E2K            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr4  <span style="text-decoration: underline;">59</span>431 A     C     hg38   Q8IYB9     ENST00000610261.6 E2A            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr4  <span style="text-decoration: underline;">59</span>431 A     G     hg38   Q8IYB9     ENST00000610261.6 E2G            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr4  <span style="text-decoration: underline;">59</span>431 A     T     hg38   Q8IYB9     ENST00000610261.6 E2V            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr4  <span style="text-decoration: underline;">59</span>432 A     C     hg38   Q8IYB9     ENST00000610261.6 E2D            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr4  <span style="text-decoration: underline;">59</span>432 A     T     hg38   Q8IYB9     ENST00000610261.6 E2D            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr4  <span style="text-decoration: underline;">59</span>433 C     A     hg38   Q8IYB9     ENST00000610261.6 L3I            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr4  <span style="text-decoration: underline;">59</span>433 C     T     hg38   Q8IYB9     ENST00000610261.6 L3F            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr4  <span style="text-decoration: underline;">59</span>433 C     G     hg38   Q8IYB9     ENST00000610261.6 L3V            </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: am_pathogenicity &lt;dbl&gt;, am_class &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="working-with-bioconductor">Working with Bioconductor<a class="anchor" aria-label="anchor" href="#working-with-bioconductor"></a>
</h2>
<p>This section illustrates how AlphaMissense data can be integrated
with other <em>Bioconductor</em> workflows, including the <a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a>
infrastructure and the <a href="https://bioconductor.org/packages/ensembldb" class="external-link">ensembldb</a> package
/ <a href="https://bioconductor.org/packages/AnnotationHub" class="external-link">AnnotationHub</a>
resources.</p>
<div class="section level3">
<h3 id="genomic-ranges">Genomic ranges<a class="anchor" aria-label="anchor" href="#genomic-ranges"></a>
</h3>
<p>The <a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a>
infrastructure provides standard data structures that allow range-based
operations across Bioconductor packages. The <code>GPos</code> data
structure provides a convenient and memory-efficient representation of
single-nucleotide variants like those in the <code>hg19</code> and
<code>hg38</code> AlphaMissense resources. Start by installing (if
necessary) the <a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a>
package.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"GenomicRanges"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GenomicRanges"</span><span class="op">)</span></span></code></pre></div>
<p>Select the <code>hg38</code> data resource, and filter to a subset of
variants; <code>GPos</code> is an in-memory data structure and can
easily manage 10’s of millions of variants.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="../reference/AlphaMissense.html">am_data</a></span><span class="op">(</span><span class="st">"hg38"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">CHROM</span> <span class="op">==</span> <span class="st">"chr2"</span>, <span class="va">POS</span> <span class="op">&lt;</span> <span class="fl">10000000</span>, <span class="va">REF</span> <span class="op">==</span> <span class="st">"G"</span><span class="op">)</span></span></code></pre></div>
<p>Use <code><a href="../reference/GenomicRanges.html">to_GPos()</a></code> to coerce to a <code>GPos</code>
object.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gpos</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">tbl</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/GenomicRanges.html">to_GPos</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">gpos</span></span>
<span><span class="co">#&gt; UnstitchedGPos object with 26751 positions and 7 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames       pos strand |         REF         ALT  uniprot_id</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt; &lt;integer&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]     chr2     41613      * |           G           C      Q1W6H9</span></span>
<span><span class="co">#&gt;       [2]     chr2     41615      * |           G           C      Q1W6H9</span></span>
<span><span class="co">#&gt;       [3]     chr2     41615      * |           G           A      Q1W6H9</span></span>
<span><span class="co">#&gt;       [4]     chr2     41615      * |           G           T      Q1W6H9</span></span>
<span><span class="co">#&gt;       [5]     chr2     41618      * |           G           C      Q1W6H9</span></span>
<span><span class="co">#&gt;       ...      ...       ...    ... .         ...         ...         ...</span></span>
<span><span class="co">#&gt;   [26747]     chr2   9999028      * |           G           A      Q9NZI5</span></span>
<span><span class="co">#&gt;   [26748]     chr2   9999028      * |           G           T      Q9NZI5</span></span>
<span><span class="co">#&gt;   [26749]     chr2   9999029      * |           G           C      Q9NZI5</span></span>
<span><span class="co">#&gt;   [26750]     chr2   9999029      * |           G           A      Q9NZI5</span></span>
<span><span class="co">#&gt;   [26751]     chr2   9999029      * |           G           T      Q9NZI5</span></span>
<span><span class="co">#&gt;                transcript_id protein_variant am_pathogenicity          am_class</span></span>
<span><span class="co">#&gt;                  &lt;character&gt;     &lt;character&gt;        &lt;numeric&gt;       &lt;character&gt;</span></span>
<span><span class="co">#&gt;       [1]  ENST00000327669.5           R321G           0.0848     likely_benign</span></span>
<span><span class="co">#&gt;       [2]  ENST00000327669.5           S320C           0.0946     likely_benign</span></span>
<span><span class="co">#&gt;       [3]  ENST00000327669.5           S320F           0.1593     likely_benign</span></span>
<span><span class="co">#&gt;       [4]  ENST00000327669.5           S320Y           0.1429     likely_benign</span></span>
<span><span class="co">#&gt;       [5]  ENST00000327669.5           P319R           0.0765     likely_benign</span></span>
<span><span class="co">#&gt;       ...                ...             ...              ...               ...</span></span>
<span><span class="co">#&gt;   [26747] ENST00000324907.14           G581R           0.9997 likely_pathogenic</span></span>
<span><span class="co">#&gt;   [26748] ENST00000324907.14           G581W           0.9996 likely_pathogenic</span></span>
<span><span class="co">#&gt;   [26749] ENST00000324907.14           G581A           0.9970 likely_pathogenic</span></span>
<span><span class="co">#&gt;   [26750] ENST00000324907.14           G581E           0.9995 likely_pathogenic</span></span>
<span><span class="co">#&gt;   [26751] ENST00000324907.14           G581V           0.9994 likely_pathogenic</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 25 sequences (1 circular) from hg38 genome</span></span></code></pre></div>
<p>Vignettes in the <a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a>
package illustrate use of these objects.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes</a></span><span class="op">(</span><span class="st">"GenomicRanges"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="grch38-annotation-resources">GRCh38 annotation resources<a class="anchor" aria-label="anchor" href="#grch38-annotation-resources"></a>
</h3>
<p>Start by identifying the most recent EnsDb resource for <em>Homo
Sapiens</em>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hub</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">hub</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"EnsDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 26 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2024-04-02</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: EnsDb</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH53211"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                             </span></span>
<span><span class="co">#&gt;   AH53211  | Ensembl 87 EnsDb for Homo Sapiens </span></span>
<span><span class="co">#&gt;   AH53715  | Ensembl 88 EnsDb for Homo Sapiens </span></span>
<span><span class="co">#&gt;   AH56681  | Ensembl 89 EnsDb for Homo Sapiens </span></span>
<span><span class="co">#&gt;   AH57757  | Ensembl 90 EnsDb for Homo Sapiens </span></span>
<span><span class="co">#&gt;   AH60773  | Ensembl 91 EnsDb for Homo Sapiens </span></span>
<span><span class="co">#&gt;   ...        ...                               </span></span>
<span><span class="co">#&gt;   AH104864 | Ensembl 107 EnsDb for Homo sapiens</span></span>
<span><span class="co">#&gt;   AH109336 | Ensembl 108 EnsDb for Homo sapiens</span></span>
<span><span class="co">#&gt;   AH109606 | Ensembl 109 EnsDb for Homo sapiens</span></span>
<span><span class="co">#&gt;   AH113665 | Ensembl 110 EnsDb for Homo sapiens</span></span>
<span><span class="co">#&gt;   AH116291 | Ensembl 111 EnsDb for Homo sapiens</span></span>
<span></span>
<span><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="st">"AH113665"</span><span class="op">]</span></span>
<span><span class="co">#&gt; AnnotationHub with 1 record</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2024-04-02</span></span>
<span><span class="co">#&gt; # names(): AH113665</span></span>
<span><span class="co">#&gt; # $dataprovider: Ensembl</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: EnsDb</span></span>
<span><span class="co">#&gt; # $rdatadateadded: 2023-04-25</span></span>
<span><span class="co">#&gt; # $title: Ensembl 110 EnsDb for Homo sapiens</span></span>
<span><span class="co">#&gt; # $description: Gene and protein annotations for Homo sapiens based on Ensem...</span></span>
<span><span class="co">#&gt; # $taxonomyid: 9606</span></span>
<span><span class="co">#&gt; # $genome: GRCh38</span></span>
<span><span class="co">#&gt; # $sourcetype: ensembl</span></span>
<span><span class="co">#&gt; # $sourceurl: http://www.ensembl.org</span></span>
<span><span class="co">#&gt; # $sourcesize: NA</span></span>
<span><span class="co">#&gt; # $tags: c("110", "Annotation", "AnnotationHubSoftware", "Coverage",</span></span>
<span><span class="co">#&gt; #   "DataImport", "EnsDb", "Ensembl", "Gene", "Protein", "Sequencing",</span></span>
<span><span class="co">#&gt; #   "Transcript") </span></span>
<span><span class="co">#&gt; # retrieve record with 'object[["AH113665"]]'</span></span></code></pre></div>
<p>Load the <a href="https://bioconductor.org/packages/ensembldb" class="external-link">ensembldb</a> library
and retrieve the record. Unfortunately, there are many conflicts between
function names in different packages, so it becomes necessary to fully
resolve functions to the package where they are defined.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jorainer/ensembldb" class="external-link">ensembldb</a></span><span class="op">)</span></span>
<span><span class="va">edb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">"AH113665"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">edb</span></span>
<span><span class="co">#&gt; EnsDb for Ensembl:</span></span>
<span><span class="co">#&gt; |Backend: SQLite</span></span>
<span><span class="co">#&gt; |Db type: EnsDb</span></span>
<span><span class="co">#&gt; |Type of Gene ID: Ensembl Gene ID</span></span>
<span><span class="co">#&gt; |Supporting package: ensembldb</span></span>
<span><span class="co">#&gt; |Db created by: ensembldb package from Bioconductor</span></span>
<span><span class="co">#&gt; |script_version: 0.3.10</span></span>
<span><span class="co">#&gt; |Creation time: Mon Aug  7 09:02:07 2023</span></span>
<span><span class="co">#&gt; |ensembl_version: 110</span></span>
<span><span class="co">#&gt; |ensembl_host: 127.0.0.1</span></span>
<span><span class="co">#&gt; |Organism: Homo sapiens</span></span>
<span><span class="co">#&gt; |taxonomy_id: 9606</span></span>
<span><span class="co">#&gt; |genome_build: GRCh38</span></span>
<span><span class="co">#&gt; |DBSCHEMAVERSION: 2.2</span></span>
<span><span class="co">#&gt; |common_name: human</span></span>
<span><span class="co">#&gt; |species: homo_sapiens</span></span>
<span><span class="co">#&gt; | No. of genes: 71440.</span></span>
<span><span class="co">#&gt; | No. of transcripts: 278545.</span></span>
<span><span class="co">#&gt; |Protein data available.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="from-bioconductor-to-duckdb">From <em>Bioconductor</em> to <em>DuckDB</em><a class="anchor" aria-label="anchor" href="#from-bioconductor-to-duckdb"></a>
</h3>
<p>In this section we will work more directly with the database,
including writing temporary tables. For illustration purposes, we use a
connection that can read and write; in practice, read-only permissions
are sufficient for creating temporary tables.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">db_rw</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/db.html">db_connect</a></span><span class="op">(</span>read_only <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>As an illustration, use <a href="https://bioconductor.org/packages/ensembldb" class="external-link">ensembldb</a> to
identify the exons of the canonical transcript of a particular gene.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bcl2l11</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">edb</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">ensembldb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ensembldb/man/global-filters.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>        <span class="op">~</span> <span class="va">symbol</span> <span class="op">==</span> <span class="st">"BCL2L11"</span> <span class="op">&amp;</span></span>
<span>            <span class="va">tx_biotype</span> <span class="op">==</span> <span class="st">"protein_coding"</span> <span class="op">&amp;</span></span>
<span>            <span class="va">tx_is_canonical</span> <span class="op">==</span> <span class="cn">TRUE</span></span>
<span>    <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcriptsBy.html" class="external-link">exonsBy</a></span><span class="op">(</span><span class="st">"tx"</span><span class="op">)</span></span>
<span><span class="va">bcl2l11</span></span>
<span><span class="co">#&gt; GRangesList object of length 1:</span></span>
<span><span class="co">#&gt; $ENST00000393256</span></span>
<span><span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames              ranges strand |         exon_id exon_rank</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;   [1]        2 111120914-111121188      + | ENSE00004011574         1</span></span>
<span><span class="co">#&gt;   [2]        2 111123733-111124139      + | ENSE00001008808         2</span></span>
<span><span class="co">#&gt;   [3]        2 111150044-111150147      + | ENSE00003483971         3</span></span>
<span><span class="co">#&gt;   [4]        2 111164133-111168444      + | ENSE00001924953         4</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from GRCh38 genome</span></span></code></pre></div>
<p>Munge the data to a tibble, updating the <code>seqnames</code> to a
column <code>CHROM</code> with identifiers such as <code>"chr1"</code>
(as in the AlphaMissense data). Write the tibble to a temporary table
(it will be deleted when <code>db</code> is disconnected from the
database) so that it can be used in ‘lazy’ SQL queries with the
AlphaMissense data.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bcl2l11_tbl</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">bcl2l11</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>CHROM <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="va">seqnames</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">CHROM</span>, <span class="fu"><a href="https://tidyselect.r-lib.org/reference/everything.html" class="external-link">everything</a></span><span class="op">(</span><span class="op">)</span>, <span class="op">-</span><span class="va">seqnames</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/db.html">db_temporary_table</a></span><span class="op">(</span><span class="va">db_rw</span>, <span class="va">bcl2l11_tbl</span>, <span class="st">"bcl2l11"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;bcl2l11&gt; [4 x 9]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;   CHROM group group_name          start       end width strand exon_id exon_rank</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> chr2      1 ENST00000393256 111<span style="text-decoration: underline;">120</span>914 111<span style="text-decoration: underline;">121</span>188   275 +      ENSE00…         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> chr2      1 ENST00000393256 111<span style="text-decoration: underline;">123</span>733 111<span style="text-decoration: underline;">124</span>139   407 +      ENSE00…         2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> chr2      1 ENST00000393256 111<span style="text-decoration: underline;">150</span>044 111<span style="text-decoration: underline;">150</span>147   104 +      ENSE00…         3</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> chr2      1 ENST00000393256 111<span style="text-decoration: underline;">164</span>133 111<span style="text-decoration: underline;">168</span>444  <span style="text-decoration: underline;">4</span>312 +      ENSE00…         4</span></span></code></pre></div>
<p>The temporary table is now available on the <code>db_rw</code>
connection; the tables will be removed on disconnect,
<code>db_disconnect(db_rw)</code>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="st">"bcl2l11"</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="../reference/db.html">db_tables</a></span><span class="op">(</span><span class="va">db_rw</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>Use <code><a href="../reference/db.html">db_range_join()</a></code> to join the AlphaMissense data with
the ranges defining the exons in our gene of interest. The arguments are
the database connection, the AlphaMissense table of interest (this table
must have columns <code>CHROM</code> and <code>POS</code>), the table
containing ranges of interest (with columns <code>CHROM</code>,
<code>start</code>, <code>end</code>), and the temporary table to
contain the results. A range join is like a standard database join,
expect that the constraints can be relations, in our case that
<code>POS &gt;= start</code> and <code>POS &lt;= end</code> for each
range of interest; implementation details are in this <a href="https://duckdb.org/2022/05/27/iejoin.html" class="external-link">DuckDB blog</a>. The
range join uses closed intervals (the start and end positions are
included in the query), following <em>Bioconductor</em> convention.
Writing to a temporary table avoids bringing potentially large datasets
into R memory, and makes the table available for subsequent manipulation
in the current session.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rng</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/db.html">db_range_join</a></span><span class="op">(</span><span class="va">db_rw</span>, <span class="st">"hg38"</span>, <span class="st">"bcl2l11"</span>, <span class="st">"bcl2l11_overlaps"</span><span class="op">)</span></span>
<span><span class="va">rng</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   table&lt;bcl2l11_overlaps&gt; [?? x 18]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;    CHROM       POS REF   ALT   genome uniprot_id transcript_id   protein_variant</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> chr2  111<span style="text-decoration: underline;">164</span>133 G     A     hg38   O43521     ENST0000039325… V167I          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> chr2  111<span style="text-decoration: underline;">164</span>133 G     T     hg38   O43521     ENST0000039325… V167L          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> chr2  111<span style="text-decoration: underline;">164</span>133 G     C     hg38   O43521     ENST0000039325… V167L          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> chr2  111<span style="text-decoration: underline;">164</span>134 T     C     hg38   O43521     ENST0000039325… V167A          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> chr2  111<span style="text-decoration: underline;">164</span>134 T     A     hg38   O43521     ENST0000039325… V167E          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> chr2  111<span style="text-decoration: underline;">164</span>134 T     G     hg38   O43521     ENST0000039325… V167G          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> chr2  111<span style="text-decoration: underline;">164</span>136 T     A     hg38   O43521     ENST0000039325… F168I          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> chr2  111<span style="text-decoration: underline;">164</span>136 T     C     hg38   O43521     ENST0000039325… F168L          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> chr2  111<span style="text-decoration: underline;">164</span>136 T     G     hg38   O43521     ENST0000039325… F168V          </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> chr2  111<span style="text-decoration: underline;">164</span>137 T     G     hg38   O43521     ENST0000039325… F168C          </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 10 more variables: am_pathogenicity &lt;dbl&gt;, am_class &lt;chr&gt;, group &lt;int&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   group_name &lt;chr&gt;, start &lt;int&gt;, end &lt;int&gt;, width &lt;int&gt;, strand &lt;fct&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   exon_id &lt;chr&gt;, exon_rank &lt;int&gt;</span></span></span></code></pre></div>
<p>This query takes place almost instantly. A larger query of 71M
variants against 1000 ranges took about 20 seconds.</p>
<p>The usual database and <a href="https://cran.r-project.org/package=dplyr" class="external-link">dplyr</a> verbs can be
used to summarize the results, e.g., the number of variants in each
pathogenecity class in each exon.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">rng</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">exon_id</span>, <span class="va">am_class</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="st">"am_class"</span>, values_from <span class="op">=</span> <span class="st">"n"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Source:   SQL [3 x 4]</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Database: DuckDB v0.10.0 [ma38727@Darwin 23.4.0:R 4.4.0//Users/ma38727/Library/Caches/org.R-project.R/R/BiocFileCache/a3a01776fe5e_a3a01776fe5e]</span></span></span>
<span><span class="co">#&gt;   exon_id         likely_benign likely_pathogenic ambiguous</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> ENSE00003483971           128                71        33</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> ENSE00001924953           172                16        19</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> ENSE00001008808           494               257       108</span></span></code></pre></div>
<p>It is perhaps instructive to review the range join as a source of
inspiration for other computations that might be of interest.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="co">-- range join of 'hg38' with 'bcl2l11'; overwrite any existing table</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a><span class="co">-- 'bcl2l11_overlaps'</span></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="kw">DROP</span> <span class="kw">TABLE</span> <span class="cf">IF</span> <span class="kw">EXISTS</span> bcl2l11_overlaps;</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a><span class="kw">CREATE</span> TEMP <span class="kw">TABLE</span> bcl2l11_overlaps <span class="kw">AS</span></span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a><span class="kw">SELECT</span></span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>    hg38.<span class="op">*</span>,</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>    bcl2l11.<span class="op">*</span> EXCLUDE (CHROM)</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a><span class="kw">FROM</span> hg38</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a><span class="kw">JOIN</span> bcl2l11</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a>    <span class="kw">ON</span> bcl2l11.CHROM <span class="op">=</span> hg38.CHROM</span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>    <span class="kw">AND</span> bcl2l11.<span class="kw">start</span> <span class="op">&lt;=</span> hg38.POS</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>    <span class="kw">AND</span> bcl2l11.<span class="cf">end</span> <span class="op">&gt;=</span> hg38.POS;</span></code></pre></div>
<p>As best practice, disconnect from the writable database connection
when work is complete.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/db.html">db_disconnect</a></span><span class="op">(</span><span class="va">db_rw</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="from-duckdb-to-bioconductor">From <em>DuckDB</em> to <em>Bioconductor</em><a class="anchor" aria-label="anchor" href="#from-duckdb-to-bioconductor"></a>
</h3>
<p>There are likely more straight-forward ways of performing the query
in the previous section, e.g., by filtering <code>hg38</code> on the
relevant transcript id(s), retrieving to <em>R</em>, and working with
<code>edb</code> to classify variants by exon. The transcript we are
interested in is <code>"ENST00000393256"</code>.</p>
<p>Select the relevant variants and, because there are not too many,
load into <em>R</em>.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">variants_of_interest</span> <span class="op">&lt;-</span></span>
<span>    <span class="fu"><a href="../reference/AlphaMissense.html">am_data</a></span><span class="op">(</span><span class="st">"hg38"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">transcript_id</span> <span class="op">%like%</span> <span class="st">"ENST00000393256%"</span><span class="op">)</span></span></code></pre></div>
<p>Coerce the AlphaMissense data to a <code>GRanges::GPos</code>
object.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gpos</span> <span class="op">&lt;-</span></span>
<span>    <span class="va">variants_of_interest</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="../reference/GenomicRanges.html">to_GPos</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## make gpos 'genome' and 'seqlevels' like bcl2l11</span></span>
<span><span class="fu">GenomeInfoDb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">genome</a></span><span class="op">(</span><span class="va">gpos</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"GRCh38"</span></span>
<span><span class="fu">GenomeInfoDb</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqlevelsStyle.html" class="external-link">seqlevelsStyle</a></span><span class="op">(</span><span class="va">gpos</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"Ensembl"</span></span>
<span><span class="va">gpos</span></span>
<span><span class="co">#&gt; UnstitchedGPos object with 1298 positions and 7 metadata columns:</span></span>
<span><span class="co">#&gt;          seqnames       pos strand |         REF         ALT  uniprot_id</span></span>
<span><span class="co">#&gt;             &lt;Rle&gt; &lt;integer&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1]        2 111123749      * |           G           C      O43521</span></span>
<span><span class="co">#&gt;      [2]        2 111123749      * |           G           T      O43521</span></span>
<span><span class="co">#&gt;      [3]        2 111123749      * |           G           A      O43521</span></span>
<span><span class="co">#&gt;      [4]        2 111123750      * |           C           A      O43521</span></span>
<span><span class="co">#&gt;      [5]        2 111123750      * |           C           G      O43521</span></span>
<span><span class="co">#&gt;      ...      ...       ...    ... .         ...         ...         ...</span></span>
<span><span class="co">#&gt;   [1294]        2 111164227      * |           A           G      O43521</span></span>
<span><span class="co">#&gt;   [1295]        2 111164227      * |           A           T      O43521</span></span>
<span><span class="co">#&gt;   [1296]        2 111164227      * |           A           C      O43521</span></span>
<span><span class="co">#&gt;   [1297]        2 111164228      * |           T           A      O43521</span></span>
<span><span class="co">#&gt;   [1298]        2 111164228      * |           T           G      O43521</span></span>
<span><span class="co">#&gt;              transcript_id protein_variant am_pathogenicity          am_class</span></span>
<span><span class="co">#&gt;                &lt;character&gt;     &lt;character&gt;        &lt;numeric&gt;       &lt;character&gt;</span></span>
<span><span class="co">#&gt;      [1] ENST00000393256.8             A2P           0.6610 likely_pathogenic</span></span>
<span><span class="co">#&gt;      [2] ENST00000393256.8             A2S           0.3398     likely_benign</span></span>
<span><span class="co">#&gt;      [3] ENST00000393256.8             A2T           0.7219 likely_pathogenic</span></span>
<span><span class="co">#&gt;      [4] ENST00000393256.8             A2E           0.7283 likely_pathogenic</span></span>
<span><span class="co">#&gt;      [5] ENST00000393256.8             A2G           0.3883         ambiguous</span></span>
<span><span class="co">#&gt;      ...               ...             ...              ...               ...</span></span>
<span><span class="co">#&gt;   [1294] ENST00000393256.8           H198R           0.1226     likely_benign</span></span>
<span><span class="co">#&gt;   [1295] ENST00000393256.8           H198L           0.2526     likely_benign</span></span>
<span><span class="co">#&gt;   [1296] ENST00000393256.8           H198P           0.1323     likely_benign</span></span>
<span><span class="co">#&gt;   [1297] ENST00000393256.8           H198Q           0.1781     likely_benign</span></span>
<span><span class="co">#&gt;   [1298] ENST00000393256.8           H198Q           0.1781     likely_benign</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 25 sequences (1 circular) from GRCh38 genome</span></span></code></pre></div>
<p>One can then use <a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a>
functionality, e.g., to count the number of variants in each exon.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">countOverlaps</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">bcl2l11</span><span class="op">)</span>, <span class="va">gpos</span><span class="op">)</span></span>
<span><span class="co">#&gt; ENST00000393256 ENST00000393256 ENST00000393256 ENST00000393256 </span></span>
<span><span class="co">#&gt;               0             859             232             207</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="finally">Finally<a class="anchor" aria-label="anchor" href="#finally"></a>
</h2>
<p>Remember to disconnect and shutdown all managed DuckDB
connections.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/db.html">db_disconnect_all</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Database connections that are not closed correctly trigger warning
messages.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2024-03-18 r86148)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin23.3.0</span></span>
<span><span class="co">#&gt; Running under: macOS Sonoma 14.4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Users/ma38727/bin/R-devel/lib/libRblas.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Users/ma38727/bin/R-devel/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/New_York</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] ensembldb_2.27.1        AnnotationFilter_1.27.0 GenomicFeatures_1.55.4 </span></span>
<span><span class="co">#&gt;  [4] AnnotationDbi_1.65.2    Biobase_2.63.0          GenomicRanges_1.55.3   </span></span>
<span><span class="co">#&gt;  [7] GenomeInfoDb_1.39.9     IRanges_2.37.1          S4Vectors_0.41.4       </span></span>
<span><span class="co">#&gt; [10] BiocGenerics_0.49.1     AlphaMissenseR_0.99.21  dplyr_1.1.4            </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.2.2                   bitops_1.0-7               </span></span>
<span><span class="co">#&gt;   [3] httr2_1.0.0                 biomaRt_2.59.1             </span></span>
<span><span class="co">#&gt;   [5] rlang_1.1.3                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [7] matrixStats_1.2.0           compiler_4.4.0             </span></span>
<span><span class="co">#&gt;   [9] RSQLite_2.3.5               png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [11] systemfonts_1.0.6           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [13] stringr_1.5.1               ProtGenerics_1.35.4        </span></span>
<span><span class="co">#&gt;  [15] pkgconfig_2.0.3             crayon_1.5.2               </span></span>
<span><span class="co">#&gt;  [17] fastmap_1.1.1               dbplyr_2.4.0               </span></span>
<span><span class="co">#&gt;  [19] XVector_0.43.1              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [21] Rsamtools_2.19.4            rmarkdown_2.26             </span></span>
<span><span class="co">#&gt;  [23] ragg_1.3.0                  purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [25] bit_4.0.5                   xfun_0.43                  </span></span>
<span><span class="co">#&gt;  [27] zlibbioc_1.49.3             cachem_1.0.8               </span></span>
<span><span class="co">#&gt;  [29] jsonlite_1.8.8              progress_1.2.3             </span></span>
<span><span class="co">#&gt;  [31] blob_1.2.4                  DelayedArray_0.29.9        </span></span>
<span><span class="co">#&gt;  [33] BiocParallel_1.37.1         parallel_4.4.0             </span></span>
<span><span class="co">#&gt;  [35] spdl_0.0.5                  prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt;  [37] R6_2.5.1                    bslib_0.6.2                </span></span>
<span><span class="co">#&gt;  [39] stringi_1.8.3               rtracklayer_1.63.1         </span></span>
<span><span class="co">#&gt;  [41] jquerylib_0.1.4             Rcpp_1.0.12                </span></span>
<span><span class="co">#&gt;  [43] SummarizedExperiment_1.33.3 knitr_1.45                 </span></span>
<span><span class="co">#&gt;  [45] BiocBaseUtils_1.5.1         Matrix_1.6-5               </span></span>
<span><span class="co">#&gt;  [47] tidyselect_1.2.1            abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [49] yaml_2.3.8                  codetools_0.2-19           </span></span>
<span><span class="co">#&gt;  [51] curl_5.2.1                  rjsoncons_1.2.0.9800       </span></span>
<span><span class="co">#&gt;  [53] lattice_0.22-5              tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [55] withr_3.0.0                 KEGGREST_1.43.0            </span></span>
<span><span class="co">#&gt;  [57] evaluate_0.23               desc_1.4.3                 </span></span>
<span><span class="co">#&gt;  [59] BiocFileCache_2.11.1        xml2_1.3.6                 </span></span>
<span><span class="co">#&gt;  [61] Biostrings_2.71.4           pillar_1.9.0               </span></span>
<span><span class="co">#&gt;  [63] BiocManager_1.30.22         filelock_1.0.3             </span></span>
<span><span class="co">#&gt;  [65] MatrixGenerics_1.15.0       whisker_0.4.1              </span></span>
<span><span class="co">#&gt;  [67] generics_0.1.3              RCurl_1.98-1.14            </span></span>
<span><span class="co">#&gt;  [69] BiocVersion_3.19.1          hms_1.1.3                  </span></span>
<span><span class="co">#&gt;  [71] glue_1.7.0                  lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [73] tools_4.4.0                 AnnotationHub_3.11.1       </span></span>
<span><span class="co">#&gt;  [75] BiocIO_1.13.0               GenomicAlignments_1.39.4   </span></span>
<span><span class="co">#&gt;  [77] fs_1.6.3                    XML_3.99-0.16.1            </span></span>
<span><span class="co">#&gt;  [79] grid_4.4.0                  tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [81] GenomeInfoDbData_1.2.11     RcppSpdlog_0.0.16          </span></span>
<span><span class="co">#&gt;  [83] duckdb_0.10.0               restfulr_0.0.15            </span></span>
<span><span class="co">#&gt;  [85] cli_3.6.2                   rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt;  [87] textshaping_0.3.7           fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [89] S4Arrays_1.3.6              sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [91] digest_0.6.35               SparseArray_1.3.4          </span></span>
<span><span class="co">#&gt;  [93] rjson_0.2.21                memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [95] htmltools_0.5.8.9000        pkgdown_2.0.7              </span></span>
<span><span class="co">#&gt;  [97] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt;  [99] mime_0.12                   bit64_4.0.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Martin Morgan, Chan Zuckerberg Initiative DAF CZF2019-002443, NIH NCI ITCR U24CA180996, NIH NCI IOTN U24CA232979, NIH NCI ARTNet U24CA274159.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
